src/OpenSpaceToolkit/Physics/Environment/Atmospheric/Earth/NRLMSISE00.cpp

```diff
@@ -185,23 +194,19 @@ Unique<NRLMSISE00::nrlmsise_input> NRLMSISE00::computeNRLMSISE00Input(
     Real lst = Real::Undefined();
 
     // Use actual sun position to compute local solar time if provided
-    if (aSunPosition.isDefined())
+    if (sunCelestialSPtr_)
```

Comment:
While this is a bit simpler, it makes it harder to test as we can't provide the sunPosition as an input...

Reply:
I don't think it's much harder to test. We just have to initialize two different atmosphere models (one with sun and one without), then compare the difference.

Reply:
I meant that before, we could actually use the sunposition from other sources (GMAT, Orekit, STK etc.). Now we have can't bootstrap that.

Reply:
Ah yep that's fair. It is harder to validate against another provider when interested specifically in how our computed sun position differs from theirs. 
With the data from yesterday (Thursday), I was able to validate that there is no difference in computed density whether we use our sun position or Orekit's. Which make me more confident that we won't need to do that again. (i.e. we're both just using the sun position provided by SPICE). 
So IMO it's worth the simplicity to add the Sun at the constructor level instead of the function level. I suppose if we need to compare the sun position again, we could mock `getPositionIn` function within the Sun class.